### PR TITLE
Set up CI for bob / rn compatibility matrix and handle non-standard codegen output dirs

### PR DESCRIPTION
**Pull request by Johennes**

### CODE REVIEW

**Comment by jhugman** [comment]
Quick drive by, I know it's in draft, just a quick note to say:

This is stunning. It's going to be so helpful. Thank you.

**Comment by Johennes** [comment]
I think this is in a state where it can be reviewed now.

I thought a bit about adding further dimensions to the test matrix such as org-scoped vs. non-org-scoped package name and standard vs. custom codegen output paths. However, these would make the number of combinations and CI jobs soar.

Instead I just used bob version, RN version and platform for the test matrix dimensions and _always_ use an org-scoped package name and custom codegen output paths. The thinking there is that if e.g. the job succeeds with custom codegen paths, it is likely to also succeed with the default ones whereas the opposite is not true.

The jobs currently all fail because of #161.

I would suggest to also run the workflow on PRs (as it is configured right now) but to not make it block the PR. This can be achieved by tweaking the required status checks in the repository settings.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [![CI](https://github.com/jhugman/uniffi-bindgen-react-native/actions/workflows/ci.yml/badge.svg)](https://github.com/jhugman/uniffi-bindgen-react-native/actions/workflows/ci.yml)
+[![build-bob compatibility](https://github.com/jhugman/uniffi-bindgen-react-native/actions/workflows/compat.yml/badge.svg)](https://github.com/jhugman/uniffi-bindgen-react-native/actions/workflows/compat.yml)
```
Comment (jhugman) [inline README.md:2]: 😻

File: scripts/test-turbo-modules.sh [hunk 2]
```diff
@@ -342,19 +341,9 @@ build_ios_example() {
...
-  if [ "$udid" == "null" ]; then
-    udid=$(xcrun simctl list --json devices | jq -r '.devices[][] | select(.isAvailable == true) | .udid' | head -n 1)
-    xcrun simctl boot "$udid"
-  fi
-
-  if [ "$udid" == "null" ]; then
-    error "No available iOS simulator found"
-  fi
-
-  xcodebuild -workspace "${IOS_NAME}Example.xcworkspace" -scheme "${IOS_NAME}Example" -configuration Debug -destination "id=$udid" || error "Failed to build iOS example"
```
Comment (jhugman) [inline scripts/test-turbo-modules.sh:357]: Oh, I see how you're getting rid of IOS_NAME now.

**Review by jhugman — APPROVED** [review]
This is really nice. Thank you. 

I will let you be the arbiter of ordering you land it in. Getting it all in order with #161 may need the build to break for a little bit, but I'm okay with that.

> I thought a bit about adding further dimensions to the test matrix such as org-scoped vs. non-org-scoped package name and standard vs. custom codegen output paths. However, these would make the number of combinations and CI jobs soar.

Yes, agreed. There are some lighter weight checks in `test-turbo-modules.sh` which might be worth separating out in another PR. These just make sure the identifiers inn the ubrn generated files match those in the builder-bob versions.

File: .github/workflows/compat.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+name: Compat
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
```
[the remark this answers is not in the document]
Comment (Johennes) [reply .github/workflows/compat.yml:5]: Wait, my cron-fu is very rusty but isn't this 00:00 every day? I certainly did not intend it to run hourly. 😅

File: scripts/test-turbo-modules.sh [hunk 4]
```diff
@@ -11,10 +11,10 @@ reset_args() {
   RN_VERSION=latest
   PROJECT_SLUG=my-test-library
   FORCE_NEW_DIR=false
-  IOS_NAME=MyTestLibrary
   SKIP_IOS=true
   SKIP_ANDROID=true
   UBRN_CONFIG=
+  PKG_JSON_MIXIN=
```
[the remark this answers is not in the document]
Comment (Johennes) [reply scripts/test-turbo-modules.sh]: Only one character more than `KEEP_ROOT_ON_EXIT` so I guess we can indeed afford it, yes. 😅

Have made the change.

**Comment by Johennes** [comment]
> I will let you be the arbiter of ordering you land it in. Getting it all in order with https://github.com/jhugman/uniffi-bindgen-react-native/issues/161 may need the build to break for a little bit, but I'm okay with that.

Thanks. I guess since this is fairly well isolated, I'll let it sit for a while in hopes that I can figure out #165 and then merge both together.

File: .github/workflows/compat.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jhugman) [reply .github/workflows/compat.yml:5]: Oh ha. I think you're right. Definitely needs a comment here I think.

[the remark this answers is not in the document]
Comment (Johennes) [reply .github/workflows/compat.yml]: Yes, agreed. Have added one.

**Comment by Johennes** [comment]
@jhugman I've merged the other PR into this one after addressing your feedback from there. Sorry for the multi-PR spam. 🙈

**Review by jhugman — APPROVED** [review]
I think if you land this now and we can fix #167 separately.

Good work, @Johennes this is a really helpful PR.

**Pull request merged**